app/lib/analyzer/task_sources.dart

```diff
@@ -77,14 +77,17 @@ class DatastoreHistoryTaskSource implements TaskSource {
   Stream<Task> startStreaming() async* {
     for (;;) {
       try {
-        await for (PackageVersion pv in _db.query(PackageVersion).run()) {
-          await new Future.delayed(const Duration(seconds: 30));
+        final Query query = _db.query(PackageVersion)..order('-created');
+        await for (PackageVersion pv in query.run()) {
           final List<PackageVersionAnalysis> list = await _db.lookup([
             _db.emptyKey
                 .append(PackageAnalysis, id: pv.package)
                 .append(PackageVersionAnalysis, id: pv.version)
           ]);
           if (list.first == null) {
+            // Waiting to throttle the historical tasks. If there is a newly
+            // uploaded package, give it a chance before continuing with these.
+            await new Future.delayed(const Duration(seconds: 30));
```

Comment:
I'm not sure that an additional class and redirection will make it easier to follow what is happening. I'd rather focus on making the Stream merger part to handle this (either by priorities or doing the throttling there). `package:async`'s `StreamQueue` could help there a lot, but it lacks the synchronous check whether there is an item waiting in the queue.